### PR TITLE
Addon-docs: Apply list styles over reset

### LINKED ISSUE #10270: Can't render list items when using MDX
## Description

I have a list of items in my `MDX` files, as defined below:

```
### Items

1. Item A.
1. Item B.
1. Item C.
```

**Storybook renders it as:**

image

**Expected behavior:**

image

**Additional context**

Looks like `list-style` has been disabled by [default](https://github.com/storybookjs/storybook/blob/953c43bec6e5ecf09689d1d86fcc01fee5eb11f4/docs/src/components/Docs/Container/style.css#L11). Is there a workaround available that I can use?

### PR DESCRIPTION
**Pull request by frassinier**
Closes #10270 

## What I did

Force a default `list-style` for un.ordered lists with addon-docs.
Tailwind or Bulma, for example, used to remove default style so let's define it, to be sure we display lists if these CSS frameworks are globally included.

## How to test

- Is this testable with Jest or Chromatic screenshots? It should not change anything 🤞 
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

### CODE REVIEW

**Review by shilman — COMMENTED** [review]
How to test?

File: lib/components/src/typography/DocumentFormatting.tsx [hunk 1]
```diff
@@ -302,6 +302,12 @@ export const LI = styled.li<{}>(withReset, ({ theme }) => ({
     marginTop: '.25em',
     marginBottom: 0,
   },
+  '& ul': {
+    listStyle: 'disc',
+  },
+  '& ol': {
+    listStyle: 'decimal',
+  },
```
Comment (ndelangen) [inline lib/components/src/typography/DocumentFormatting.tsx:310]: This causes:
https://www.chromatic.com/snapshot?appId=5a375b97f4b14f0020b0cda3&id=5ef26a5d350d7d0022a27d4f

But I'm fine with it actually

**Comment by frassinier** [comment]
@shilman [here](https://5a375b97f4b14f0020b0cda3-cubdcdqttm.chromatic.com/?path=/docs/docs-docspage--markdown) is a link with a story for which I reset CSS style to `list-style: none` like some of CSS frameworks used to do.

**Comment by shilman** [comment]
cc @domyen can you take a look?

**Comment by domyen** [comment]
lgtm

**Review by shilman — APPROVED** [review]
LGTM per @domyen 's review

**Pull request merged**